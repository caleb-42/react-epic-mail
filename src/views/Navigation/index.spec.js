import React from "react";
import renderer from "react-test-renderer";
import { cleanup, render } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import configureStore from "@redux/configureStore";
import Nav from "./index";

const store = configureStore();

const FakeNav = (args) => {
  return render(
    <ReduxProvider store={store}>
      <Nav {...args} className="" />
    </ReduxProvider>
  )
}

describe('SNAPSHOT TEST', () => {

  it("Nav Component with SideBar", () => {
    const tree = renderer.create(
      <ReduxProvider store={store}>
        <Nav position="side" />
      </ReduxProvider >
    );
    expect(tree).toMatchSnapshot();
  });

  it("Nav Component with TopBar", () => {
    const tree = renderer.create(
      <ReduxProvider store={store}>
        <Nav position="top" />
      </ReduxProvider >
    );
    expect(tree).toMatchSnapshot();
  });
})

describe('COMPONENT TEST', () => {
  /* 
    afterEach(cleanup);
  
    it("should render menu name in Header", () => {
      const { getByText } = FakeHeader();
      getByText('Mails');
    });
  
    it("should render subMenu name in Header", () => {
      const { getByText } = FakeHeader({ activeNav: { menu: 'Mails', subMenu: 'Inbox' } });
      getByText('Inbox');
    }); */
})
