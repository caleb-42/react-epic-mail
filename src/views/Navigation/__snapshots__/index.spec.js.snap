// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT TEST Nav Component with SideBar 1`] = `
<aside
  className="appNav side-nav overflow-y-auto h-screen bg-gray-800 w-1/5"
>
  <div
    className="logo"
  />
  <ul
    className="side-nav mt-16 relative navig anim"
  >
    <li
      className=""
    >
      <h3
        className="active nav text-white px-12 py-6 text-xl font-semibold"
        onClick={[Function]}
      >
        Mails
      </h3>
      <ul
        className="mail-types bg-gray-750 anim w-100 open"
      >
        <li
          className="active subnav px-12 py-4 text-md"
          onClick={[Function]}
        >
          Inbox
        </li>
        <li
          className="text-white subnav px-12 py-4 text-md"
          onClick={[Function]}
        >
          Unread
        </li>
        <li
          className="text-white subnav px-12 py-4 text-md"
          onClick={[Function]}
        >
          Sent
        </li>
        <li
          className="text-white subnav px-12 py-4 text-md"
          onClick={[Function]}
        >
          Draft
        </li>
      </ul>
    </li>
    <li
      className="pointer-events-none opacity-50"
    >
      <h3
        className=" nav text-white px-12 py-6 text-xl font-semibold"
        onClick={[Function]}
      >
        Groups
      </h3>
    </li>
    <li
      className="pointer-events-none opacity-50"
    >
      <h3
        className=" nav text-white px-12 py-6 text-xl font-semibold"
        onClick={[Function]}
      >
        Settings
      </h3>
    </li>
    <li
      className=""
    >
      <h3
        className=" nav text-white px-12 py-6 text-xl font-semibold"
        onClick={[Function]}
      >
        Sign Out
      </h3>
    </li>
  </ul>
</aside>
`;

exports[`SNAPSHOT TEST Nav Component with TopBar 1`] = `
<ul
  className="appNav top-nav hide navig w-full anim bg-gray-800"
>
  <li>
    <h3
      className="active nav text-white px-12 py-6 text-xl font-semibold"
      onClick={[Function]}
    >
      Mails
    </h3>
    <ul
      className="mail-types bg-gray-750 anim w-100 open"
    >
      <li
        className="active subnav px-12 py-4 text-md"
        onClick={[Function]}
      >
        Inbox
      </li>
      <li
        className="text-white subnav px-12 py-4 text-md"
        onClick={[Function]}
      >
        Unread
      </li>
      <li
        className="text-white subnav px-12 py-4 text-md"
        onClick={[Function]}
      >
        Sent
      </li>
      <li
        className="text-white subnav px-12 py-4 text-md"
        onClick={[Function]}
      >
        Draft
      </li>
    </ul>
  </li>
  <li>
    <h3
      className=" nav text-white px-12 py-6 text-xl font-semibold"
      onClick={[Function]}
    >
      Groups
    </h3>
  </li>
  <li>
    <h3
      className=" nav text-white px-12 py-6 text-xl font-semibold"
      onClick={[Function]}
    >
      Settings
    </h3>
  </li>
  <li>
    <h3
      className=" nav text-white px-12 py-6 text-xl font-semibold"
      onClick={[Function]}
    >
      Sign Out
    </h3>
  </li>
</ul>
`;
